tests/basic/storage_pool_test.go

```diff
@@ -5296,29 +5298,32 @@ var _ = Describe("{PoolResizeVolumesResync}", func() {
 
 			log.InfoD("setting replication on the volumes")
 			setRepl := func(vol *volume.Volume) error {
+				var (
+					maxReplicaFactor int64
+					nodesToBeUpdated []string
+					poolsToBeUpdated []string
+				)
+				maxReplicaFactor = 3
+				nodesToBeUpdated = nil
+				poolsToBeUpdated = nil
+
 				log.InfoD("setting replication factor of the volume [%v] with ID [%v]", vol.Name, vol.ID)
 				currRepFactor, err := Inst().V.GetReplicationFactor(vol)
 				log.FailOnError(err, "Failed to get replication factor on the volume")
 				log.Infof("Replication factor on the volume [%v] is [%v]", vol.Name, currRepFactor)
 				opts := volume.Options{
 					ValidateReplicationUpdateTimeout: replicationUpdateTimeout,
 				}
+				poolsToBeUpdated = append(poolsToBeUpdated, rebootPoolID)
 				if currRepFactor == 3 {
 					newRepl := currRepFactor - 1
-					err = Inst().V.SetReplicationFactor(vol, newRepl, nil, nil, true, opts)
+					err = Inst().V.SetReplicationFactor(vol, newRepl, nodesToBeUpdated, poolsToBeUpdated, true, opts)
```

Comment:
here u should reduce the volume replication on poolsToBeUpdated only if volume has replication on it. is it passing even if volume has no replication on it and pool id is passed?

Reply:
Done